pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -311,9 +312,26 @@ func (c *Controller) syncNetworkWithGNP(ctx context.Context, network *networkv1.
 	return nil
 }
 
-func (c *Controller) handleGKENetworkParamSetDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
+func (c *Controller) handleGKENetworkParamSetDelete(ctx context.Context, params *networkv1.GKENetworkParamSet, key string) error {
+	if params.Status.NetworkName == "" {
```

Comment:
we may describe this better :

- removeFinalizer IF `params.Status.NetworkName == ""` or the NetworkName no longer exist or the network is not in Use (whatever this in use means. ;)
- retry after X if the network is in Use() describe this better and the reasoning about that
- ..

and then I feel we can improve code readability and remove some of the nested checks, but is a feeling didn't check it carefully and is not a blocker

Reply:
I was able to simplify this a little by removing the requeue.